src/Microsoft.Identity.Web.DownstreamApi/DownstreamApi.cs

```diff
@@ -38,15 +45,18 @@ internal partial class DownstreamApi : IDownstreamApi
         /// <param name="namedDownstreamApiOptions">Named options provider.</param>
         /// <param name="httpClientFactory">HTTP client factory.</param>
         /// <param name="logger">Logger.</param>
+        /// <param name="serviceProvider">The service provider.</param>
         public DownstreamApi(
             IAuthorizationHeaderProvider authorizationHeaderProvider,
             IOptionsMonitor<DownstreamApiOptions> namedDownstreamApiOptions,
             IHttpClientFactory httpClientFactory,
-            ILogger<DownstreamApi> logger)
+            ILogger<DownstreamApi> logger,
+            IServiceProvider serviceProvider)

```

Comment:
You'd want to avoid the locator pattern. Rather add a constructor with the IMsalHttpClientFactory? and keep this one?

Reply:
Good suggestion, will do.